### PR TITLE
Implement link proxy links

### PR DESCRIPTION
**Pull request by tusbar**
### Tasks

- [x] Use `got` instead of `superagent` in order to retry requests.
- [x] Merge wfs and file downloads in a single array of resources.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,18 +42,20 @@
...
     "plunger": "^0.1.3",
+    "pump": "^3.0.0",
     "request": "^2.79.0",
     "require-all": "^2.2.0",
     "rimraf": "^2.4.2",
     "superagent": "^3.5.2",
     "tldjs": "^2.0.0",
     "tmp": "0.0.31",
     "underscore.string": "^3.0",
+    "unzipper": "^0.9.3",
```
Comment (jdesboeufs) [inline package.json:57]: Why not `decompress` ou `yauzl`? (It's just a question)

File: package.json [hunk 2]
```diff
@@ -40,18 +42,20 @@
     "mongoose": "^5.0.2",
     "morgan": "^1.5.1",
     "natural": "^0.5.4",
-    "ogr2ogr": "jdesboeufs/ogr2ogr",
+    "ogr2ogr": "^1.3.0",
     "on-finished": "^2.3.0",
     "passport": "^0.4.0",
     "passport-oauth2": "^1.4.0",
     "plunger": "^0.1.3",
+    "pump": "^3.0.0",
```
Comment (jdesboeufs) [inline package.json:50]: Duplicates with `mississippi` but I prefer explicit dependencies :)

File: lib/api/routes/links.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        code: 404,
+        message: `Download with id ${req.params.downloadId} was not found`
+      })
+    }
+
+    const tmp = await createTempDirectory()
+    const path = join(tmp, download.name)
+
+    if (download.archive) {
+      await unzipFile(download.url, tmp)
```
Comment (jdesboeufs) [inline lib/api/routes/links.js:68]: Rejection will never be caught

File: lib/api/routes/links.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    } else {
+      await downloadFile(download.url, path)
```
Comment (jdesboeufs) [inline lib/api/routes/links.js:70]: Same

File: lib/jobs/consolidate-record/links.js [hunk 5]
```diff
@@ -1,111 +1,95 @@
...
+  table: [
+    'csv',
+    'xls'
+  ],
+  data: [
+    'grib2',
+    'owl',
+    'rdf',
+    'kml',
+    'gpx',
```
Comment (jdesboeufs) [inline lib/jobs/consolidate-record/links.js]: gpx and kml are vectors

File: package.json (hunk 1 above)
Comment (tusbar) [reply package.json:57]: `unzipper` seemed more straighforward. `yauzl` is too complicated for just unzipping files in the file system.

Also `unzipper` supports streaming.

```js
function unzipFile(url, tmp) {
  return new Promise((resolve, reject) => {
    pump(
      got.stream(url),
      new unzipper.Extract({
        path: tmp
      }),
      error => {
        if (error) {
          return reject(error)
        }

        resolve()
      }
    )
  })
```

`decompress` seems nice.

File: package.json (hunk 2 above)
Comment (tusbar) [reply package.json:50]: Me too, also, `mississippi` is too hard to type.

File: lib/api/routes/links.js (hunk 3 above)
Comment (tusbar) [reply lib/api/routes/links.js:68]: it will now

**Pull request merged**